pinot-core/src/main/java/com/linkedin/pinot/core/realtime/stream/StreamConfig.java

```diff
@@ -16,172 +16,207 @@
-      if (key.startsWith(decoderPropPrefix)) {
-        _decoderProperties.put(StreamConfigProperties.getPropertySuffix(key, decoderPropPrefix),
-            streamConfigMap.get(key));
+    int flushThresholdRows = DEFAULT_FLUSH_THRESHOLD_ROWS;
+    String flushThresholdRowsValue = streamConfigMap.get(StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_ROWS);
+    if (flushThresholdRowsValue != null) {
+      try {
+        flushThresholdRows = Integer.parseInt(flushThresholdRowsValue);
+      } catch (Exception e) {
+        LOGGER.warn("Caught exception when parsing flush threshold rows {}:{}, defaulting to {}",
+            StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_ROWS, flushThresholdRowsValue, DEFAULT_FLUSH_THRESHOLD_ROWS,
+            e);
       }
+    }
+    _flushThresholdRows = flushThresholdRows;
 
-      // TODO: Move kafkaConsumerProp to kafka specific StreamConfig
-      String kafkaConsumerPropPrefix =
-          KafkaStreamConfigProperties.constructStreamProperty(KafkaStreamConfigProperties.KAFKA_CONSUMER_PROP_PREFIX);
-      if (key.startsWith(kafkaConsumerPropPrefix)) {
-        _kafkaConsumerProperties.put(StreamConfigProperties.getPropertySuffix(key, kafkaConsumerPropPrefix),
-            streamConfigMap.get(key));
+    long flushThresholdTime = DEFAULT_FLUSH_THRESHOLD_TIME;
+    String flushThresholdTimeValue = streamConfigMap.get(StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_TIME);
+    if (flushThresholdTimeValue != null) {
+      try {
+        flushThresholdTime = TimeUtils.convertPeriodToMillis(flushThresholdTimeValue);
+      } catch (Exception e) {
+        LOGGER.warn("Caught exception when converting flush threshold period to millis {}:{}, defaulting to {}",
+            StreamConfigProperties.SEGMENT_FLUSH_THRESHOLD_TIME, flushThresholdTimeValue, DEFAULT_FLUSH_THRESHOLD_TIME,
+            e);
       }
     }
+    _flushThresholdTimeMillis = flushThresholdTime;
+
+    long flushDesiredSize = -1;
+    String flushSegmentDesiredSizeValue = streamConfigMap.get(StreamConfigProperties.SEGMENT_FLUSH_DESIRED_SIZE);
+    if (flushSegmentDesiredSizeValue != null) {
+      flushDesiredSize = DataSize.toBytes(flushSegmentDesiredSizeValue);
+    }
+    if (flushDesiredSize > 0) {
+      _flushSegmentDesiredSizeBytes = flushDesiredSize;
+    } else {
+      _flushSegmentDesiredSizeBytes = DEFAULT_DESIRED_SEGMENT_SIZE_BYTES;
+    }
+
+    _streamConfigMap.putAll(streamConfigMap);
+  }
+
+  public String getType() {
+    return _type;
+  }
+
+  public String getTopicName() {
+    return _topicName;
+  }
+
+  public List<ConsumerType> getConsumerTypes() {
+    return _consumerTypes;
   }
 
   public boolean hasHighLevelConsumerType() {
     return _consumerTypes.contains(ConsumerType.HIGHLEVEL);
   }
 
   public boolean hasLowLevelConsumerType() {
-    return _consumerTypes.contains(ConsumerType.SIMPLE);
+    return _consumerTypes.contains(ConsumerType.LOWLEVEL);
```

Comment:
Maybe as a followup (or in this PR), we can change these methods to hasStreamLevelConsumerType() and hasPartitionLevelConsumerType() ?

Reply:
Will do this in a followup PR. Those method name changes will touch too many files, and it better go in separately